model/common/src/icon4py/model/common/metrics/metrics_factory.py

```diff
@@ -0,0 +1,554 @@
+from icon4py.model.common.test_utils.helpers import constant_field
+
+
+dt_utils.TEST_DATA_ROOT = pathlib.Path(__file__).parent / "testdata"
+properties = decomposition.get_processor_properties(decomposition.get_runtype(with_mpi=False))
+path = dt_utils.get_datapath_for_experiment(
+    dt_utils.get_ranked_data_path(dt_utils.SERIALIZED_DATA_PATH, properties)
+)
+
+data_provider = sb.IconSerialDataProvider(
+    "icon_pydycore", str(path.absolute()), False, mpi_rank=properties.rank
+)
+
+# z_ifc (computable from vertical grid for model without topography)
+metrics_savepoint = data_provider.from_metrics_savepoint()
+
+# interpolation fields also for now passing as precomputed fields
+interpolation_savepoint = data_provider.from_interpolation_savepoint()
+# can get geometry fields as pre computed fields from the grid_savepoint
+root, level = dt_utils.get_global_grid_params(dt_utils.REGIONAL_EXPERIMENT)
+grid_id = dt_utils.get_grid_id_for_experiment(dt_utils.REGIONAL_EXPERIMENT)
+grid_savepoint = data_provider.from_savepoint_grid(grid_id, root, level)
+nlev = grid_savepoint.num(dims.KDim)
+cell_domain = horizontal.domain(dims.CellDim)
+edge_domain = horizontal.domain(dims.EdgeDim)
+vertex_domain = horizontal.domain(dims.VertexDim)
+#######
+
+# start build up factory:
+
+# used for vertical domain below: should go away once vertical grid provids start_index and end_index like interface
+grid = grid_savepoint.global_grid_params
+
+interface_model_height = metrics_savepoint.z_ifc()
+c_lin_e = interpolation_savepoint.c_lin_e()
+k_index = gtx.as_field((dims.KDim,), xp.arange(nlev + 1, dtype=gtx.int32))
+vct_a = grid_savepoint.vct_a()
+theta_ref_mc = metrics_savepoint.theta_ref_mc()
+exner_ref_mc = metrics_savepoint.exner_ref_mc()
+wgtfac_c = metrics_savepoint.wgtfac_c()
+c_refin_ctrl = grid_savepoint.refin_ctrl(dims.CellDim)
+e_refin_ctrl = grid_savepoint.refin_ctrl(dims.EdgeDim)
+dual_edge_length = grid_savepoint.dual_edge_length()
+tangent_orientation = grid_savepoint.tangent_orientation()
+inv_primal_edge_length = grid_savepoint.inverse_primal_edge_lengths()
+inv_dual_edge_length = grid_savepoint.inv_dual_edge_length()
+cells_aw_verts = interpolation_savepoint.c_intp().asnumpy()
+cells_aw_verts_field = gtx.as_field((dims.VertexDim, dims.V2CDim), cells_aw_verts)
+vwind_offctr = 0.2
+icon_grid = grid_savepoint.construct_icon_grid(on_gpu=False)
+vwind_impl_wgt_full = constant_field(icon_grid, 0.5 + vwind_offctr, dims.CellDim)
+experiment = dt_utils.GLOBAL_EXPERIMENT
+init_val = 0.65 if experiment == dt_utils.GLOBAL_EXPERIMENT else 0.7
+vwind_impl_wgt_k = constant_field(icon_grid, init_val, dims.CellDim, dims.KDim)
+k_lev = gtx.as_field((dims.KDim,), np.arange(nlev, dtype=gtx.int32))
+e_lev = gtx.as_field((dims.EdgeDim,), np.arange(icon_grid.num_edges, dtype=gtx.int32))
+e_owner_mask = grid_savepoint.e_owner_mask()
+
+fields_factory = factory.FieldsFactory()
+
+fields_factory.register_provider(
+    factory.PrecomputedFieldsProvider(
+        {
```

Comment:
That is probably fine as it is a workaround anyway. In general, hopefully we can make it such that the precomputed ones are only grid geometry fields and interpolation fields (what is for the test data in the grid_savepoint and the interpolation_savepoint as they should come from another source. (The grid manager or an interpolation fields factory) 
I guess we will have to reduce the ones that are here one after the other.